### ISSUE TITLE
Automatic Func<T> registration not working 

### ISSUE THREAD

**Issue by j-hn** [unitycontainer/unity#304]
We've just updated a project to Unity 5.11, and unity abstractions from 3.3.1 to 5.11.1
the automatic registration of Func<RegisteredType> is no longer working. is there a configuration that needs setting?
Lazy<T> seems to work, but i dont want to have to change all our constructors..

```C#
example classes to illustrate
   public class Pudgy2
    {
      
    }
    public class Pudgy
    {
        private readonly Pudgy2 _pudgy2;
        public Pudgy2 InnerPudgy => _pudgy2;
        public Pudgy(Func<Pudgy2> pudgy)
        {
            _pudgy2 = pudgy();
        }
    
        public  Pudgy()
        { }

    }

the tests
```C#
//This test fails to resolve the Factory Method
    [Test]
        public void Test1FailsToResolve()
        {
            var container=new UnityContainer();
            container.RegisterType<Pudgy>();
            container.RegisterType<Pudgy2>();
            var pudgy=container.Resolve<Pudgy>();
            Assert.IsNull(pudgy.InnerPudgy);
        }
//this test manually registers the factory method, and resolves
        [Test]
        public void TestResolves()
        {
            var container = new UnityContainer();
            container.RegisterType<Pudgy>();
            container.RegisterType<Pudgy2>();
            container.RegisterType<Func<Pudgy2>>();
            var pudgy = container.Resolve<Pudgy>();
            Assert.IsNotNull(pudgy.InnerPudgy);
        }


cheers John

**Comment by ENikS** [comment]
Did you try to run it with Diagnostic extension? What is the error?

**Comment by j-hn** [comment]
There is no error as such. (I ran with the diagnostic extension but no error was reported)
In our system we have some large classes with multiple constructor parameters, several of which are Func<T>. I noticed the issue as the new unity creates the classes using the empty constructor, and not the constructor with the parameters. 
on investigation if i removed Func<T> params it works, calling the constructor with the most parameters, as soon as these are introduced back it goes back to calling the empty constructor.
if I change them to Lazy<T> then it also works - but i dont really want to have to change the constructor params.
thanks
John
